suite-common/wallet-utils/src/accountUtils.ts

```diff
@@ -365,22 +364,26 @@ export const formatTokenAmount = (tokenTransfer: TokenTransfer) => {
     return formattedTokenSymbol ? `${formattedAmount} ${formattedTokenSymbol}` : formattedAmount;
 };
 
-export const sortByCoin = (accounts: Account[]) =>
-    accounts.sort((a, b) => {
-        const aIndex = networksCompatibility.findIndex(n => {
-            const accountType = n.accountType || ACCOUNT_TYPE.NORMAL;
+export const sortByCoin = (accounts: Account[]) => {
+    const orderedNetworkSymbols = Object.keys(networks) as NetworkSymbol[];
 
-            return accountType === a.accountType && n.symbol === a.symbol;
-        });
-        const bIndex = networksCompatibility.findIndex(n => {
-            const accountType = n.accountType || ACCOUNT_TYPE.NORMAL;
+    return accounts.sort((a, b) => {
+        // sort by order of networks key (symbols)
+        const aSymbolIndex = orderedNetworkSymbols.indexOf(a.symbol);
+        const bSymbolIndex = orderedNetworkSymbols.indexOf(b.symbol);
+        if (aSymbolIndex !== bSymbolIndex) return aSymbolIndex - bSymbolIndex;
 
-            return accountType === b.accountType && n.symbol === b.symbol;
-        });
-        if (aIndex === bIndex) return a.index - b.index;
+        // same network, so sort by accountType keys
+        const network = networks[a.symbol];
+        const orderedAccountTypes = Object.keys(network.accountTypes) as AccountType[];
+        const aAccountTypeIndex = orderedAccountTypes.indexOf(a.accountType);
+        const bAccountTypeIndex = orderedAccountTypes.indexOf(b.accountType);
```

Comment:
This is new, isn't it? Is going to be used somewhere? If not, I wouldn't do it.

Reply:
Yes, it is necessary to keep the behavior same. Let me illustrate:

**Before, with `networksCompatibility`**:
```
networksCompatibility: 
- btc, normal
- btc, taproot
- eth, normal
- eth, ledger
```
We need to sort by network as primary criterion, and by account as secondary (that's what the old fn did).
It was a simpler implementation, because `networksCompatibility` is a flat structure already sorted exactly the way we as we want, so we just look up the indices.

**Now with `networks`** the nested structure, I really have to do two steps. First sort by network (btc vs. eth, but normal vs. taproot are not yet sorted), and then by account (sort btc normal vs .btc taproot).

Still, I tried to remove the four lines, and test coverage fails :x: